veps/sig-storage/utility-volumes.md

```diff
@@ -0,0 +1,256 @@
+# VEP \#86: Utility Volumes for virt-launcher Pods
+
+## Release Signoff Checklist
+
+Items marked with (R) are required *prior to targeting to a milestone / release*.
+
+  - [+] (R) Enhancement issue created, which links to VEP dir in [kubevirt/enhancements] (not the initial VEP PR)
+  - [ ] (R) Target version is explicitly mentioned and approved
+  - [ ] (R) Graduation criteria filled
+
+## Overview
+
+This proposal introduces a new API mechanism within KubeVirt to define **hot-pluggable, maintainence-focused volumes** that are attached directly to the `virt-launcher` Pod, rather than being exposed to the Virtual Machine (VM) itself. These `utilityVolumes` will facilitate various out-of-band operations such as collecting backup output, performing diagnostic data collection like memory-dump, logs, and other future operational needs. This design generalizes the concept currently present in the `memorydump` volume source, providing a more versatile and extensible solution for similar dynamic, host-level needs.
+
+## Motivation
+
+KubeVirt currently offers the `memorydump` volume source, which, upon request, hot-attaches a PersistentVolumeClaim (PVC) to the `virt-launcher` Pod to store a memory dump of the VM. While powerful for its specific use case, there is a broader need for similar **temporary, hot-pluggable, host-level data collection** that doesn't involve mounting inside the VM's guest OS. For example:
+
+  * **Backup Data Collection:** This proposal is a **required foundational component for the [Incremental Backup VEP](https://github.com/kubevirt/enhancements/blob/main/veps/sig-storage/incremental-backup.md)**. The Incremental Backup VEP necessitates a mechanism to **dynamically attach a PVC** to the `virt-launcher` Pod to collect the backup output directly from libvirt. This allows for efficient data transfer and management of backup artifacts.
```

Comment:
I guess the changed block tracking would still be handled in the VM state PVC, right? Would it make sense to include it in this PVC?

Reply:
Makes sense, thanks!

Reply:
@aglitke @alromeros do you know what was the original design rationale for having the VMState PVC only in the status and not in the spec? this is one of the major difficulties in the design of this feature.. definitely if it can be changed it would be for the best.
I do stand behind what I said where the VMState PVC does not belong as a utilityVolume. Those are inherently hotplug volumes - not persistent, and does not interact with the domain at all.